itest/rust/src/builtin_tests/string/gstring_test.rs

```diff
@@ -69,12 +69,14 @@ fn string_chars() {
     // Empty tests regression from #228: Null pointer passed to slice::from_raw_parts().
     let string = GString::new();
     assert_eq!(string.chars(), &[]);
+    assert_eq!(string, GString::from(&[][..]));
```

Comment:
`&[][..]` is quite weird :thinking: please the expected variable separately and annotate with `&[char]` type.

Reply:
How about `[].as_slice()`?

Reply:
I would still add `&[char]` so it's clear that this conversion is being tested. Type erasure and compact expressions can be nice sometimes, but tests should primarily be readable and easy to understand. Any indirection to figure out that we're looking at a _char_ slice (not just a slice) is an unnecessary obstacle.

And once you have that, `[].as_slice()` involves arrays without a very good reason, so you could just have `&[]` 😉

Reply:
To be fair; doesn't `&[]` involve arrays, too? :thinking:

Reply:
You're right, that was a bit imprecise. What I meant is that `&[1, 2, 3]` is very common syntax, but `[1, 2, 3].as_slice()` makes readers think "what is this?"; same for empty ones :slightly_smiling_face: 

Also, sorry to be nitpicky, but I wrote "variable separately" above. You did an `as` cast instead.
```rs
assert_eq!(string, GString::from(&[] as &[char]));
```
Please change this to
```rs
let empty: &[char] = &[];
assert_eq!(string, GString::from(empty));
```
Again because people may wonder "why cast a type into itself" and lints may actually flag this in the future.